src/OpenTelemetry/Trace/SamplingParameters.cs

```diff
@@ -42,9 +45,11 @@ namespace OpenTelemetry.Trace
             ActivityTraceId traceId,
             string name,
             ActivityKind kind,
-            IEnumerable<KeyValuePair<string, object>> tags = null,
-            IEnumerable<ActivityLink> links = null)
+            IEnumerable<KeyValuePair<string, object?>>? tags = null,
+            IEnumerable<ActivityLink>? links = null)
         {
+            Guard.ThrowIfNullOrWhitespace(name);
```

Comment:
This is kind of interesting. OTel spec says span name is required. But...

```csharp
myActivitySource.StartActivity(name: null);
```

...is allowed.

@tarekgh 

```csharp
        [Fact]
        public void ActivityWithNullName()
        {
            using ActivitySource s = new("my source");
            using ActivityListener l = new();
            l.ShouldListenTo = (n) => true;
            l.Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllDataAndRecorded;
            ActivitySource.AddActivityListener(l);

            using var activity = s.StartActivity(name: null); // <- This should throw?

            Assert.NotNull(activity);
            Assert.Null(activity.DisplayName);
            Assert.Null(activity.OperationName);
        }
```

Seems like we are missing some validation? `Activity.DisplayName` & `Activity.OperationName` are decorated as non-nullable.

Reply:
This is a legacy behavior introduced before OTel exists. https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Activity.cs,443
We didn't use to throw when passing null to the Activity constructor. So null activity name is allowed

Reply:
Going to resolve this conversation for this PR. Might make other changes as a follow-up. For this work, I just coalesce `null` span name to `string.Empty` on `SamplingParameters`. That prevents leakage of the `null` into our OTel API. @alanwest had the suggestion to automatically discard anything with `null` span name. Seems reasonable. But we can do that as its own thing. Opened #3802 to track.